src/NativeScript/Calling/FFICache.h

```diff
@@ -0,0 +1,45 @@
+//
+//  FFICache.h
+//  NativeScript
+//
+//  Created by Teodor Dermendzhiev on 02/10/2017.
+//
+//
+
+#ifndef __NativeScript__FFICache__
+#define __NativeScript__FFICache__
+
+#include "FFIType.h"
+#include <vector>
+
+namespace NativeScript {
+    
+    struct SignatureHash {
+        
+        std::size_t operator()(std::vector<const ffi_type*> signature) const {
+            std::size_t seed = 0;
+            
+            
+            for(size_t i = 0; i < signature.size(); i++) {
+                seed <<= 4;
+                seed |= signature[i]->type;
```

Comment:
I'd recommend using the [Fowler–Noll–Vo hash function](https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function). This one seems to not be distributive enough.